homeassistant/components/matter/light.py

```diff
@@ -53,30 +54,9 @@ class MatterLight(MatterEntity, LightEntity):
     """Representation of a Matter light."""
 
     entity_description: LightEntityDescription
-
-    @property
-    def supports_color(self) -> bool:
-        """Return if the device supports color control."""
-        if not self._attr_supported_color_modes:
-            return False
-        return (
-            ColorMode.HS in self._attr_supported_color_modes
-            or ColorMode.XY in self._attr_supported_color_modes
-        )
-
-    @property
-    def supports_color_temperature(self) -> bool:
-        """Return if the device supports color temperature control."""
-        if not self._attr_supported_color_modes:
-            return False
-        return ColorMode.COLOR_TEMP in self._attr_supported_color_modes
-
-    @property
-    def supports_brightness(self) -> bool:
-        """Return if the device supports bridghtness control."""
-        if not self._attr_supported_color_modes:
-            return False
-        return ColorMode.BRIGHTNESS in self._attr_supported_color_modes
+    _supports_brightness = False
+    _supports_color = False
+    _supports_color_temperature = False
```

Comment:
We don't really need these flags, we can let `self.supported_color_modes` be the source of truth instead and use light utility functions such as `light.brightness_supported(self.supported_color_modes)` when needed.